x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/FunctionRegistry.java

```diff
@@ -91,124 +92,142 @@
+                def(ACos.class, ACos::new),
+                def(ASin.class, ASin::new),
+                def(ATan.class, ATan::new),
+                def(ATan2.class, ATan2::new),
+                def(Cbrt.class, Cbrt::new),
+                def(Ceil.class, Ceil::new, "CEILING"),
+                def(Cos.class, Cos::new),
+                def(Cosh.class, Cosh::new),
+                def(Cot.class, Cot::new),
+                def(Degrees.class, Degrees::new),
+                def(E.class, E::new),
+                def(Exp.class, Exp::new),
+                def(Expm1.class, Expm1::new),
+                def(Floor.class, Floor::new),
+                def(Log.class, Log::new),
+                def(Log10.class, Log10::new),
+                // SQL and ODBC require MOD as a _function_
+                def(Mod.class, Mod::new),
+                def(Pi.class, Pi::new),
+                def(Power.class, Power::new),
+                def(Radians.class, Radians::new),
+                def(Random.class, Random::new, "RAND"),
+                def(Round.class, Round::new),
+                def(Sign.class, Sign::new, "SIGNUM"),
+                def(Sin.class, Sin::new),
+                def(Sinh.class, Sinh::new),
+                def(Sqrt.class, Sqrt::new),
+                def(Tan.class, Tan::new),
+                def(Truncate.class, Truncate::new));
+        // String
+        addToMap(def(Ascii.class, Ascii::new),
+                def(BitLength.class, BitLength::new),
+                def(Char.class, Char::new),
+                def(CharLength.class, CharLength::new, "CHARACTER_LENGTH"),
+                def(Concat.class, Concat::new),
+                def(Insert.class, Insert::new),
+                def(LCase.class, LCase::new),
+                def(Left.class, Left::new),
+                def(Length.class, Length::new),
+                def(Locate.class, Locate::new),
+                def(LTrim.class, LTrim::new),
+                def(OctetLength.class, OctetLength::new),
+                def(Position.class, Position::new),
+                def(Repeat.class, Repeat::new),
+                def(Replace.class, Replace::new),
+                def(Right.class, Right::new),
+                def(RTrim.class, RTrim::new),
+                def(Space.class, Space::new),
+                def(Substring.class, Substring::new),
+                def(UCase.class, UCase::new));
+        // Special
+        addToMap(def(Score.class, Score::new));
+    }
+    
+    private void addToMap(FunctionDefinition...functions) {
+        // temporary map to hold [function_name/alias_name : function instance]
+        Map<String, FunctionDefinition> batchMap = new HashMap<>();
         for (FunctionDefinition f : functions) {
-            defs.put(f.name(), f);
+            batchMap.put(f.name(), f);
             for (String alias : f.aliases()) {
                 Object old = aliases.put(alias, f.name());
                 if (old != null) {
```

Comment:
Maybe add also a check that the alias is not used as a primary name for another function.